gravitee-plugin-core/src/test/java/io/gravitee/plugin/core/internal/PluginRegistryTest.java

```diff
@@ -92,71 +104,127 @@ public void startTwiceWorkspace() throws Exception {
-        pluginRegistry.setEnvironment(environment);
+    @Test
+    void start_with_valid_workspace_with_duplication() throws Exception {
+        String path = "/io/gravitee/plugin/with-duplication/";
+
+        // we are using different version to assert which one is loaded
+        updateModifiedTimestampCustomPlugin2();
+
+        PluginRegistryImpl pluginRegistry = initPluginRegistry(path);
+        pluginRegistry.start();
+
+        assertThat(pluginRegistry.plugins()).hasSize(1);
+        assertThat(pluginRegistry.plugins()).first().extracting(Plugin::id).isEqualTo("custom-plugin");
+        assertThat(pluginRegistry.plugins()).first().extracting(p -> p.manifest().version()).isEqualTo("2.0.0-SNAPSHOT");
+    }
+
+    @Test
+    void start_several_workspaces_with_duplication() throws Exception {
+        String path1 = "/io/gravitee/plugin/with-duplication/";
+        String path2 = "/io/gravitee/plugin/with-dependencies/";
+
+        // need to the empty constructor thus redo all the config
+        PluginRegistryConfiguration configuration = new PluginRegistryConfiguration();
+        configuration.setPluginsPath(new String[] { getActualPath(path1), getActualPath(path2) });
+        PluginRegistryImpl pluginRegistry = new PluginRegistryImpl(configuration, environment, executor, mock(EventManager.class));
+
+        updateModifiedTimestampCustomPlugin2();
+        pluginRegistry.start();
+
+        assertThat(pluginRegistry.plugins()).hasSize(4);
+        assertThat(pluginRegistry.plugins())
+            .extracting(Plugin::id)
+            .containsExactlyInAnyOrder("my-policy-1", "my-policy-2", "my-policy-3", "custom-plugin");
+        assertThat(pluginRegistry.plugins())
+            .filteredOn(p -> p.id().equals("custom-plugin"))
+            .extracting(p -> p.manifest().version())
+            .contains("2.0.0-SNAPSHOT");
+    }
+
+    private PluginRegistryImpl initPluginRegistry(String path) {
+        PluginRegistryImpl pluginRegistry = new PluginRegistryImpl(
+            mock(PluginRegistryConfiguration.class),
+            environment,
+            executor,
+            mock(EventManager.class)
+        );
+        pluginRegistry.setWorkspacesPath(getActualPath(path));
         return pluginRegistry;
     }
+
+    private static String getActualPath(String path) {
+        URL dir = PluginRegistryTest.class.getResource(path);
+        assertThat(dir).isNotNull();
+        return URLDecoder.decode(dir.getPath(), StandardCharsets.UTF_8);
+    }
+
+    private static void updateModifiedTimestampCustomPlugin2() throws IOException, InterruptedException {
+        String path = "/io/gravitee/plugin/with-duplication/";
+        Path pluginFileOlder = Paths.get(getActualPath(path)).resolve("custom-plugin-1.0.0-SNAPSHOT.zip");
+        Path pluginFile = Paths.get(getActualPath(path)).resolve("custom-plugin-2.0.0-SNAPSHOT.zip");
+        byte[] bytes = Files.readAllBytes(pluginFile);
+        Files.write(pluginFile, bytes);
```

Comment:
Not sure I understand well but is it a way to change the last modified date?
Does `Files.setLastModifiedTime(...)` could do the job?

Reply:
If only I knew it existed, maybe I'd've used it.